### PR TITLE
Pimp my online cli

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- improves the online learning CLI
- uses table to display the history and some fancy cli library to request corrected intent / entities / actions

TODO:
- [x] needed to remove the concert online test - so currently the online learning is not tested and needs some unit tests

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by tmbo** [comment]
The code is ready for review, I'll write tests once I get to them. I guess it is not to important to review the `online.py` (nearly all of the changes in there are "UI" related) - focus should be on the other changes.

**Comment by tmbo** [comment]
This is ready for review & merge.

File: docs/interactive_learning.rst [hunk 1]
```diff
@@ -40,92 +40,100 @@ the right action before proceeding:
...
-
+reviews!) so we select that action.
 
 Now we can keep talking to the bot for as long as we like to create a longer
-conversation. At any point you can type ``0`` and the bot will write the
-current conversation to a file and exit the conversation. Make sure to
-combine the dumped story with your original training data for the next
-training.
+conversation. At any point you can press ``Ctrl-C`` and the bot will write
+provide you with exit options, e.g. writing the created conversations as
```
Comment (ricwo) [inline docs/interactive_learning.rst]: _will provide you with_

File: rasa_core/train.py [hunk 2]
```diff
@@ -199,14 +213,30 @@ def train_dialogue_model(domain_file, stories_file, output_path,
...
-                             cmdline_args.history,
-                             cmdline_args.dump_stories,
-                             additional_arguments)
+    _endpoints = AvailableEndpoints.read_endpoints(cmdline_args.endpoints)
+    _interpreter = NaturalLanguageInterpreter.create(cmdline_args.nlu,
+                                                     _endpoints.nlu)
+
+    if cmdline_args.core:
+        if not cmdline_args.online:
+            raise ValueError("--core can only be used together with "
```
Comment (ricwo) [inline rasa_core/train.py]: (with _the_ online flag)

File: tests/test_trackers.py [hunk 3]
```diff
@@ -327,7 +327,7 @@ def test_dump_and_restore_as_json(default_agent, tmpdir_factory):
     for tracker in trackers:
         out_path = tmpdir_factory.mktemp("tracker").join("dumped_tracker.json")
 
-        dumped = tracker.current_state(should_include_events=True)
+        dumped = tracker.current_state(EventVerbosity.AFTER_RESTART)
```
Comment (ricwo) [inline tests/test_trackers.py]: shall we add three quick tests for the `EventVerbosity`s `NONE`, `ALL` and `APPLIED`?

Comment (tmbo) [reply tests/test_trackers.py]: good point :+1:
